src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> oldTail = tail;
+        Node<T> newNode = new Node<>(oldTail, value, null);
+        tail = newNode;
+        if (oldTail == null) {
+            head = newNode;
+        } else {
+            oldTail.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
```

Comment:
reuse existing methods where posible and simlify if/else tree
```suggestion
        if (index == size) {
            add(value);
            return;
        } 
        checkIndex(index);
        if (index == 0) {
```